RecoLocalTracker/SiStripClusterizer/plugins/SiStripClustersToLegacy.cc

```diff
@@ -0,0 +1,161 @@
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
+#include "DataFormats/SiStripClusterSoA/interface/SiStripClusterHost.h"
+#include "DataFormats/SiStripDigiSoA/interface/SiStripDigiHost.h"
+
+// using namespace ::sistrip;
+namespace sistrip {
+  class SiStripClustersToLegacy : public edm::global::EDProducer<> {
+  public:
+    SiStripClustersToLegacy(edm::ParameterSet const& iConfig)
+        : siStripClustersToken_(consumes(iConfig.getParameter<edm::InputTag>("source"))),
+          siStripDigiToken_(consumes(iConfig.getParameter<edm::InputTag>("source"))),
+          siStripClustersSetVecPutToken_(produces()) {}
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<edm::InputTag>("source");
+      descriptions.addWithDefaultLabel(desc);
+    }
+
+    void produce(edm::StreamID, edm::Event& iEvent, edm::EventSetup const&) const override {
+      auto const& clusters_onHost = iEvent.get(siStripClustersToken_);
+      auto const& amplitudes_onHost = iEvent.get(siStripDigiToken_);
```

Comment:
No need to make changes to the code, but just to clarify for me: these are called `_onHost` - but there are no device data products involved in this module, right ?